### PR TITLE
Add Kubernetes audit acquisition

### PR DESCRIPTION
**Pull request by blotus**
- Add Kubernetes audit logs acquisition by exposing a webhook
 - Add `UnmarshalJSON`  helper to unmarshal a log line during parsing, removing the need to call `JSONExtract` each time we want to get a field from a JSON string
 - Add `GetMeta`  function in go template to allow for easier meta retrieval in notification templates
 - Update `expr` version to add support for NilSafe getters

### CODE REVIEW

**Review by wojtekxtx — CHANGES_REQUESTED** [review]
Fix errors ( see #1767 ) as well as merging conflicts

File: pkg/acquisition/modules/kubernetesaudit/k8s_audit.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+		if err != nil {
+			ka.logger.Errorf("Error marshaling audit event: %s", err)
+			continue
+		}
+		ka.logger.Tracef("Got audit event: %s", string(bytesEvent))
+		l := types.Line{
+			Raw:     string(bytesEvent),
+			Labels:  ka.config.Labels,
+			Time:    auditEvent.StageTimestamp.Time,
+			Src:     fmt.Sprintf("%s:%d", ka.config.ListenAddr, ka.config.ListenPort),
```
Comment (buixor) [inline pkg/acquisition/modules/kubernetesaudit/k8s_audit.go:183]: can we keep track of the real source ? (ip and src port for ex)

**Review by buixor — COMMENTED** [review]
minor feedback, overall look good. Question as discussed : do we want the HTTP listener to be generic now or later ?

File: pkg/acquisition/modules/kubernetesaudit/k8s_audit.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+	ka.logger.Tracef("webhookHandler called")
+	var auditEvents audit.EventList
+
+	jsonBody, err := io.ReadAll(r.Body)
+	if err != nil {
+		ka.logger.Errorf("Error reading request body: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	ka.logger.Tracef("webhookHandler receveid: %s", string(jsonBody))
```
Comment (buixor) [inline pkg/acquisition/modules/kubernetesaudit/k8s_audit.go:168]: shouldn't we have a `.Debugf` version ? Might be useful to debug the datasource ?

**Review by buixor — APPROVED** [review]
just a suggestion on having a debug level for this, otherwise LGTM

**Review by mmetc — APPROVED** [review]

**Pull request merged**